openmdao/core/driver.py

```diff
@@ -464,7 +464,8 @@ def _get_vars_to_record(self, recording_options):
 
         myoutputs = set(myoutputs)
         if recording_options['record_desvars']:
-            myoutputs.update(model.get_source(n) for n in self._designvars)
+            myoutputs.update(model.get_source(n) for n in self._designvars if model.get_source(n)
```

Comment:
It's not a huge deal, but it would be nice if we didn't call model.get_source(n) twice in the same loop.